src/net_processing.cpp

```diff
@@ -473,9 +498,26 @@ void FindNextBlocksToDownload(NodeId nodeid, unsigned int count, std::vector<con
     // Make sure pindexBestKnownBlock is up to date, we'll need it.
     ProcessBlockAvailability(nodeid);
 
+    if (!blocksToDownloadFirst.empty()) {
+        for (const PriorityBlockRequest &r: blocksToDownloadFirst) {
+            if (r.downloaded) continue;
+            if (r.pindex && state->pindexBestKnownBlock != NULL && state->pindexBestKnownBlock->nHeight >= r.pindex->nHeight && !mapBlocksInFlight.count(r.pindex->GetBlockHash())) {
```

Comment:
Split in multiple tests. Suggestion for loop body:
```cpp
if (!r.pindex) continue;
if (state->pindexBestKnownBlock == nullptr) continue;
if (state->pindexBestKnownBlock->nHeight < r.pindex->nHeight) continue;
if (mapBlocksInFlight.count(r.pindex->GetBlockHash())) continue;

vBlocks.push_back(r.pindex);

if (vBlocks.size() == count) break;
```
IMHO more readable and more easier to add comments.

Reply:
Otherwise, replace `NULL` with nullptr and use `break` instead of `return true`.